src/uct/sm/mm/mm_posix.c

```diff
@@ -79,14 +91,158 @@ static ucs_status_t uct_posix_test_mem(size_t length, int shm_fd)
     return status;
 }
 
+static size_t uct_posix_get_path_size(uct_pd_h pd)
+{
+    uct_mm_pd_t *mm_pd = ucs_derived_of(pd, uct_mm_pd_t);
+    uct_posix_pd_config_t *posix_config = ucs_derived_of(mm_pd->config,
+                                                         uct_posix_pd_config_t);
+
+    /* if shm_open is requested, the path to the backing file is /dev/shm
+     * by default. however, if shm_open isn't used, in case UCS_NO was set for
+     * use_shm_open or if UCS_TRY was set but using shm_open() was unsuccessful,
+     * the size of the path to the requested backing file is needed so that the
+     * user would know how much space to allocated for the rkey. */
+    if (posix_config->use_shm_open == UCS_YES) {
+        return 0;
+    } else {
+        return 1 + strlen(posix_config->path);
+    }
+}
+
+static ucs_status_t uct_posix_set_path(char **file_name, int use_shm_open,
+                                       const char *path, uint64_t uuid)
+{
+    ucs_status_t status;
+    int ret;
+
+    if (use_shm_open) {
+        ret = snprintf(*file_name, NAME_MAX, "/ucx_posix_mm_%s_%s_%016lx",
```

Comment:
there is code duplication here which can be avoided:
1. print the path in case it is posix
2. append the file name